--- mumudvb.orig/src/tune.c	2010-01-03 17:53:34.000000000 +0100
+++ mumudvb/src/tune.c	2012-02-19 01:43:29.431490684 +0100
@@ -37,7 +37,6 @@
 #include <sys/ioctl.h>
 #include <sys/poll.h>
 #include <unistd.h>
-#include <error.h>
 #include <errno.h>
 #include <string.h>
 
--- mumudvb.orig/src/mumudvb.c	2010-01-03 17:53:34.000000000 +0100
+++ mumudvb/src/mumudvb.c	2012-02-19 01:46:29.503143782 +0100
@@ -128,6 +128,9 @@
 #include "transcode.h"
 #endif
 
+// fix for android (lacks of gnu extensions)
+char *program_invocation_short_name = PACKAGE;
+
 /** the table for crc32 claculations */
 extern uint32_t       crc32_table[256];
 
--- mumudvb.orig/src/mumudvb.h	2012-03-03 18:53:00.882029145 +0100
+++ mumudvb/src/mumudvb.h	2012-03-03 18:53:48.017437243 +0100
@@ -96,15 +96,15 @@
 
 #define DEFAULT_PATH_LEN 256
 /**The path for the auto generated config file*/
-#define GEN_CONF_PATH "/var/run/mumudvb/mumudvb_generated_conf_card%d_tuner%d"
+#define GEN_CONF_PATH "mumudvb_generated_conf_card%d_tuner%d"
 /**The path for the list of streamed channels*/
-#define STREAMED_LIST_PATH "/var/run/mumudvb/channels_streamed_adapter%d_tuner%d"
+#define STREAMED_LIST_PATH "channels_streamed_adapter%d_tuner%d"
 /**The path for the list of *not* streamed channels*/
-#define NOT_STREAMED_LIST_PATH "/var/run/mumudvb/channels_unstreamed_adapter%d_tuner%d"
+#define NOT_STREAMED_LIST_PATH "channels_unstreamed_adapter%d_tuner%d"
 /**The path for the cam_info*/
-#define CAM_INFO_LIST_PATH "/var/run/mumudvb/caminfo_adapter%d_tuner%d"
+#define CAM_INFO_LIST_PATH "caminfo_adapter%d_tuner%d"
 /** The path for the pid file */
-#define PIDFILE_PATH "/var/run/mumudvb/mumudvb_adapter%card_tuner%tuner.pid"
+#define PIDFILE_PATH "mumudvb_adapter%card_tuner%tuner.pid"
 
 /**RTP header length*/
 #define RTP_HEADER_LEN 12
