--- mumudvb.orig/src/tune.c	2010-01-03 17:53:34.000000000 +0100
+++ mumudvb/src/tune.c	2012-02-19 01:43:29.431490684 +0100
@@ -37,7 +37,6 @@
 #include <sys/ioctl.h>
 #include <sys/poll.h>
 #include <unistd.h>
-#include <error.h>
 #include <errno.h>
 #include <string.h>
 
--- mumudvb.orig/src/mumudvb.c	2010-01-03 17:53:34.000000000 +0100
+++ mumudvb/src/mumudvb.c	2012-02-19 01:46:29.503143782 +0100
@@ -128,6 +128,9 @@
 #include "transcode.h"
 #endif
 
+// fix for android (lacks of gnu extensions)
+char *program_invocation_short_name = PACKAGE;
+
 /** the table for crc32 claculations */
 extern uint32_t       crc32_table[256];
 
